etna/packages/etna-js/utils/cancellable_helpers.jsx

```diff
@@ -1,16 +1,12 @@
 import 'regenerator-runtime';
 import {Cancellable, cancelledAsMaybe} from "./cancellable";
-import {useState, useEffect, useCallback} from 'react';
+import {useState, useEffect, useCallback, useRef} from 'react';
 
 export function useAsyncCallback(fn, deps, cleanup = () => null) {
-  const [_, setContext] = useState(() => new Cancellable());
+  const context = useRef(new Cancellable());
```

Comment:
`useRef` => very similar to useState, but crucially it is not managed in terms of render cycles and its reference is *stable*.

In some contexts, this concept is known as a "cell".  Essentially, the idea is to put a value you want to read and write into a length 1 array or an object with a single key, so that you can share that parent object's reference in order to share the mutations on the inner value.

```
a = 1
f(a)
a = 2 // f does not see the new a value, it still sees 1
```

```
a = [1]
f(a)
a[0] = 2 // f sees the updated value because it is contained within the shared reference that accepts mutations.
```

I'll explain why I use a reference instead of the state hook bellow